platform/macos/detect.py

```diff
@@ -72,7 +76,7 @@ def ver_parse(a):
     return os.path.join(os.path.join(dirname, ver_file), "MoltenVK/MoltenVK.xcframework/macos-arm64_x86_64/")
 
 
-def configure(env):
+def configure(env) -> None:
```

Comment:
I'm not used to type hints in Python but this doesn't seem to provide a lot of added value (and it's inconsistent with all other method definitions we have which don't return). I would only use type hints where they actually make things better.

Reply:
The purpose is to force mypy to check the function. When a function does not have any signature, it's not checked (see example below). 

Yes, there's a config option to check even untyped methods, but that would mean to deal with ~80 existing errors in the codebase to have it running (I already checked), so start with opt-in checks feels like a good initial step.

```
$ cat check.py
def unchecked():
    1 + "2"

def checked() -> None:
    3 + "4"

$ mypy check.py
check.py:5: error: Unsupported operand types for + ("int" and "str")
        3 + "4"
            ^
Found 1 error in 1 file (checked 1 source file)
```

Reply:
>  it's inconsistent with all other method definitions we have which don't return

if the issue here is a consistency, then the return types can be added to other functions eventually as well. Which I think is a good thing to do anyway, since it will make the scripts more robust and safer to touch/refactor.